### PR TITLE
fix: Add ellipsis to long text in Select component

### LINKED ISSUE #568: [Select] Select component does not ellipsis long values
Selecting a value with long text in the `<Select>` component does not shorten the string
## Expected Behavior

String should be shortened via some method, preferably an ellipsis.

## Current Behavior

String extends beyond the edge of the container and makes it hard to see the icons

kapture 2018-11-29 at 16 45 10

### PR DESCRIPTION
**Pull request by jonsadka**
#### Fixed Issues?

Fixes #568

#### Tests Added + Pass?

Yes

#### Any Dependency Changes?
No

#### Before

kapture 2018-11-29 at 16 45 10

#### After

kapture 2018-11-29 at 16 46 08

### CODE REVIEW

File: src/select/styled-components.js [hunk 1]
```diff
@@ -186,6 +196,7 @@ export const StyledSingleValue = styled('div', props => {
     whiteSpace: 'nowrap',
     maxWidth: '100%',
     ...getControlPadding($size, sizing, $type),
+    ...ellipsisText(),
```
Comment (chasestarr) [inline src/select/styled-components.js]: maybe in-line if the properties are not dynamic?

File: src/select/styled-components.js [hunk 2]
```diff
@@ -35,6 +35,16 @@ function getControlPadding(size = SIZE.default, sizing, type) {
   }[size];
 }
 
+function ellipsisText() {
```
Comment (blackswanny) [inline src/select/styled-components.js]: I would like to reuse in other components. We can put it here src/styles/util.js

Comment (austin94) [reply src/select/styled-components.js]: It we are going to reuse we may want to drop the `maxWidth` value

**Comment by schnerd** [comment]
Nice!

**Pull request merged**